tests/cpp/test_matmul_aten_evaluation.cpp

```diff
@@ -457,12 +470,107 @@ TEST_P(ATenNodesParametrizedTest, MatmulNodeSymbolic) {
   EXPECT_TRUE(at::allclose(out[0], out_ref));
 }
 
+TEST_P(LinearNodeParametrizedTest, LinearNodeConcrete) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  const auto& [a_shape, b_shape, bias_shape] = GetParam();
+
+  auto tv0 = makeConcreteTensor(a_shape, DataType::Half);
+  auto tv1 = makeConcreteTensor(b_shape, DataType::Half);
+  TensorView* bias = nullptr;
+  if (bias_shape.has_value()) {
+    bias = makeConcreteTensor(*bias_shape, DataType::Half);
+  }
+  auto tv2 = linear(tv0, tv1, bias);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  if (bias_shape.has_value()) {
+    fusion->addInput(bias);
+  }
+  fusion->addOutput(tv2);
+
+  at::Tensor t0 = at::randn(a_shape, at::kHalf).cuda();
+  at::Tensor t1 = at::randn(b_shape, at::kHalf).cuda();
+  std::optional<at::Tensor> bias_opt = std::nullopt;
+  if (bias_shape.has_value()) {
+    bias_opt = bias_shape.value().empty() ? at::scalar_tensor(3.14).to(at::kHalf).cuda(): at::randn(*bias_shape, at::kHalf).cuda();
+  }
+  at::Tensor out_ref = at::linear(t0, t1, bias_opt);
+
+  FusionExecutorCache fec(std::move(fusion));
+
+  std::vector<at::Tensor> out = {};
+  if (bias_shape.has_value()) {
+    out = fec.runFusionWithInputs({t0, t1, bias_opt});
+  } else {
+    out = fec.runFusionWithInputs({t0, t1});
+  }
+
+  const std::vector<FusionExecutor>& executors =
+      fec.getMostRecentKernelRuntime()->executors();
+  EXPECT_EQ(executors.size(), 1);
+  // Verify that fusion compilation was skipped.
+  EXPECT_FALSE(executors.front().hasCompiledKernel());
+
+  EXPECT_TRUE(at::allclose(out[0], out_ref));
+}
+TEST_P(LinearNodeParametrizedTest, LinearNodeSymbolic) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  const auto& [a_shape, b_shape, bias_shape] = GetParam();
+
+  auto tv0 = makeSymbolicTensor(a_shape.size(), DataType::Half);
+  auto tv1 = makeSymbolicTensor(b_shape.size(), DataType::Half);
+
+  TensorView* bias = nullptr;
+  if (bias_shape.has_value()) {
+    bias = makeSymbolicTensor(*bias_shape, DataType::Half);
+  }
+
+  auto tv2 = linear(tv0, tv1, bias);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  if (bias_shape.has_value()) {
+    fusion->addInput(bias);
+  }
+  fusion->addOutput(tv2);
+
+  at::Tensor t0 = at::randn(a_shape, at::kHalf).cuda();
+  at::Tensor t1 = at::randn(b_shape, at::kHalf).cuda();
+  std::optional<at::Tensor> bias_opt = std::nullopt;
+  if (bias_shape.has_value()) {
+    bias_opt = bias_shape.value().empty() ? at::scalar_tensor(3.14).to(at::kHalf).cuda() : at::randn(*bias_shape, at::kHalf).cuda();
```

Comment:
Does the regular randn call not work when `bias_shape` is empty?

Reply:
Had errors with binding the output of `at::tensor` and `at::randn` for empty bias shape (it was identified as rank 1). `torch.randn(0)` also creates an empty tensor instead of a scalar.

Reply:
you should be able to do that.

`torch.randn(0)` will give you an `empty` tensor instead of a scalar tensor. I think you need `torch.randn()`. instead.
Similarly, if you want to use `at::tensor`, make sure you are passing in the value as a scalar rather than a vector.

i.e. do `at::tensor(1.5)` instead of `at::tensor({1.5})`.

Reply:
FYI:
```
  auto options = at::TensorOptions()
                     .device(at::kCUDA, 0);
  std::vector<int64_t> shape = {};
  at::Tensor t6 = at::randn(shape, options.dtype(at::kFloat));
  std::cout << t6 << std::endl;
```
```
-0.924662
[ CUDAFloatType{} ]
```
This worked for me so you shouldn't need to worry about the conditional above.

Reply:
Right thanks, `torch.randn(())` generates a scalar whereas `torch.randn()` gave me an empty tensor giving the impression I needed to use another call